### PR TITLE
Feature/documents

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

File: core/documents-manager.php [hunk 1]
```diff
@@ -140,7 +144,7 @@ public function	ajax_save( $request ) {
 		$document = $this->get( $request['post_id'] );
 
 		if ( ! $document->is_built_with_elementor() || ! $document->is_editable_by_current_user() ) {
-			throw new \Exception( 'no_access' );
+			throw new \Exception( 'Access Denied' );
```
Comment (ramiy) [inline core/documents-manager.php]: See: https://github.com/pojome/elementor/pull/3282

Use: `throw new \Exception( 'Access denied.' );`

and update the phpDocs to: `@throw \Exception If current user don't have permissions to edit the post or the post is not using Elementor.`

File: core/documents-manager.php [hunk 2]
```diff
@@ -73,6 +73,10 @@ public function get( $post_id ) {
 		if ( ! isset( $this->documents[ $post_id ] ) ) {
 			$doc_type = get_post_meta( $post_id, Document::TYPE_META_KEY, true );
 
+			if ( ! $doc_type ) {
```
Comment (KingYes) [inline core/documents-manager.php]: remove

File: core/settings/page/manager.php [hunk 3]
```diff
@@ -75,7 +77,16 @@ public static function is_cpt_custom_templates_supported() {
 	 */
 	public function template_include( $template ) {
 		if ( is_singular() ) {
-			$page_template = get_post_meta( get_the_ID(), '_wp_page_template', true );
+			$post_id = get_the_ID();
+
+			if ( is_preview() || Plugin::$instance->preview->is_preview_mode() ) {
```
Comment (KingYes) [inline core/settings/page/manager.php]: doc support?

File: core/settings/page/manager.php [hunk 4]
```diff
@@ -111,7 +122,21 @@ public function get_name() {
 	 * @return BaseModel
 	 */
 	public function get_model_for_config() {
-		return $this->get_model( get_the_ID() );
+		$post_id = get_the_ID();
```
Comment (KingYes) [inline core/settings/page/manager.php]: doc support?

**Review by KingYes — CHANGES_REQUESTED** [review]

File: core/documents-manager.php [hunk 5]
```diff
@@ -82,20 +73,44 @@ public function get( $post_id ) {
...
+	 * @param string $type
 	 * @param array $post_data
 	 * @param array $meta_data
 	 *
 	 * @return Document
 	 */
-	public function create( $class_name, $post_data = [], $meta_data = [] ) {
+	public function create( $type, $post_data = [], $meta_data = [] ) {
+		if ( ! isset( $this->types[ $type ] ) ) {
+			// Translators: %s = document type name.
```
Comment (KingYes) [inline core/documents-manager.php]: /* tra

File: core/documents-manager.php [hunk 6]
```diff
@@ -82,20 +73,44 @@ public function get( $post_id ) {
...
+			wp_die( sprintf( __( 'Type %s isn\'t exist', '' ), $type ) );
+		}
+
 		if ( empty( $post_data['post_title'] ) ) {
-			$post_data['post_title'] = __( 'Elementor', '' );
+			$post_data['post_title'] = __( 'Elementor', '' ) . ' ' . ucfirst( $type );
```
Comment (KingYes) [inline core/documents-manager.php]: remove type

**Review by ramiy — APPROVED** [review]

**Pull request merged**